### PR TITLE
Proposal for downsampling

### PR DESCRIPTION
**Pull request by mhausenblas**
**What this PR does**:
This PR is WIP, it is a proposal for introducing a downsampling feature in Cortex. The proposal currently:

* Motivates the downsampling feature
* Discusses considerations and tradeoffs
* Reviews prior art and potential solutions
* Proposes requirements and recommends a solution direction

**Which issue(s) this PR fixes**:
This PR addresses #4322 

**Checklist**
- [ ] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Review by pracucci — COMMENTED** [review]
Thanks for working on this! I agree that offline downsampling seems the preferred solution. Are you also working on a more details proposed solution, on how to practically build it?

**Comment by mhausenblas** [comment]
Thanks for the initial feedback @pracucci and yes, a more detailed proposal to come, I thought maybe first a high-level convo at tomorrow's community call and when we have a rough agreement on the direction, following up with the detailed design.

File: docs/proposals/downsampling.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+### Considerations
+
+*Cardinality*: independent of the design, we expect that downsampling does not
+help with cardinality explosion, as downsampling solely works along the time
+dimension.
+
+*Accuracy*: when performing downsampling, one has two options: A) keep the raw
+samples around, that is, store the downsampled data alonside the raw data, or
+B) discard the raw data either on ingestion or after some tim and, in the long
```
Comment (jeromeinsf) [inline docs/proposals/downsampling.md:40]: nit: time

File: docs/proposals/downsampling.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+* MUST support opt-in (default behavior is as without downsampling).
+* MUST support tenant-level granularity (can opt-in on an per-tenant basis).
+* SHOULD not have a significant impact on the write path performance.
+
+Concerning the sampling tiers we want to support the following time periods:
+
+1. Data up to one week raw (not downsampled).
+1. Data older than one week up to six months downsampled to 1h intervals.
+1. Data older than six months downsampled to 24h intervals.
```
Comment (jplouis) [inline docs/proposals/downsampling.md:138]: Would this be UTC or configurable per tenant? Does timezone ever get considered in a 24h downsample?

Comment (mhausenblas) [reply docs/proposals/downsampling.md:138]: Thanks for the question! Yes, my expectation was per-tenant and will make sure to address the time zone question in the next iteration of the design.

**Comment by mhausenblas** [comment]
Sorry for the delay, I shall get back to it week after.

**Comment by wilfriedroset** [comment]
still needed please 🙏🏾

**Pull request closed without merging**